### PR TITLE
close response objects to free resources

### PR DESCRIPTION
**Pull request by ji11er**
# Description

okhttp3 declares that Response objects needs closure to free resources, this wasn't done so far.

#48 

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [x ] Code compiles correctly
* [ ] Created/updated tests
* [ x] Extended the documentation

### CODE REVIEW

File: sdk/src/main/java/io/dapr/actors/client/IActorProxy.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+ */
+package io.dapr.actors.client;
+
+import io.dapr.actors.ActorId;
+
+/**
+ *
+ * Provides the interface for implementation of proxy access for actor service.
+ */
+public interface IActorProxy {
```
Comment (artursouza) [inline sdk/src/main/java/io/dapr/actors/client/IActorProxy.java]: We don't use the I* convention. Call ActorProxy and rename the implementation to ActorProxyImpl.

File: sdk/src/main/java/io/dapr/actors/client/ActorProxy.java [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+import io.dapr.actors.communication.client.ActorRemotingClient;
+
+/**
+ * Provides the base implementation for the proxy to the remote actor objects
+ * implementing {@link #IActor} interfaces. The proxy object can be used used
+ * for client-to-actor and actor-to-actor communication.
+ *
+ * @author Swen Schisler <swen.schisler@fourtytwosoft.io>
+ */
+public class ActorProxy implements IActorProxy {
```
Comment (artursouza) [inline sdk/src/main/java/io/dapr/actors/client/ActorProxy.java]: We should not expose implementation, please make it package accessible only.

File: sdk/src/main/java/io/dapr/actors/client/ActorProxy.java [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+import io.dapr.actors.communication.IActorMessageBodyFactory;
+import io.dapr.actors.communication.client.ActorNonRemotingClient;
+import io.dapr.actors.communication.client.ActorRemotingClient;
+
+/**
+ * Provides the base implementation for the proxy to the remote actor objects
+ * implementing {@link #IActor} interfaces. The proxy object can be used used
+ * for client-to-actor and actor-to-actor communication.
+ *
+ * @author Swen Schisler <swen.schisler@fourtytwosoft.io>
```
Comment (artursouza) [inline sdk/src/main/java/io/dapr/actors/client/ActorProxy.java]: We are not adding @author in the code.

File: sdk/src/main/java/io/dapr/actors/client/IActorProxyFactory.java [hunk 4]
```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT License.
+ */
+package io.dapr.actors.client;
+
+import io.dapr.actors.ActorId;
+
+public interface IActorProxyFactory {
```
Comment (artursouza) [inline sdk/src/main/java/io/dapr/actors/client/IActorProxyFactory.java]: Please, remove this interface. We will use the Builder pattern, similar to what we have in the http client.

File: sdk/src/main/java/io/dapr/actors/communication/IActorMessageBodyFactory.java [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT License.
+ */
+package io.dapr.actors.communication;
+
+/**
+ * Defines the interface that must be implemented for providing factory for
+ * creating actor request body and response body objects.
+ */
+public interface IActorMessageBodyFactory {
```
Comment (artursouza) [inline sdk/src/main/java/io/dapr/actors/communication/IActorMessageBodyFactory.java]: We do not use the I* convention for interfaces. Just the name: https://azure.github.io/azure-sdk/java_design.html#naming-patterns

File: sdk/src/main/java/io/dapr/actors/communication/IActorResponseMessageBody.java [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+
+import java.lang.reflect.Type;
+
+/**
+ * Defines the interface that must be implemented to provide Request Message
+ * Body for remoting requests . This contains all the parameters remoting method
+ * has.
+ *
+ */
+public interface IActorResponseMessageBody {
```
Comment (artursouza) [inline sdk/src/main/java/io/dapr/actors/communication/IActorResponseMessageBody.java]: https://azure.github.io/azure-sdk/java_design.html#model-classes

File: sdk/src/main/java/io/dapr/actors/communication/IActorRequestMessageBody.java [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+package io.dapr.actors.communication;
+
+import java.lang.reflect.Type;
+
+/**
+ * Defines the interface that must be implemented to provide Request Message
+ * Body for remoting requests . This contains all the parameters remoting method
+ * has.
+ */
+public interface IActorRequestMessageBody {
```
Comment (artursouza) [inline sdk/src/main/java/io/dapr/actors/communication/IActorRequestMessageBody.java]: No need for interface in model classes: https://azure.github.io/azure-sdk/java_design.html#model-classes

File: sdk/src/main/java/io/dapr/actors/communication/IActorMessageBodyFactory.java (hunk 5 above)
Comment (artursouza) [reply sdk/src/main/java/io/dapr/actors/communication/IActorMessageBodyFactory.java]: No builder or factory for model classes: https://azure.github.io/azure-sdk/java_design.html#model-classes

File: sdk/src/main/java/io/dapr/actors/client/IActorProxyFactory.java (hunk 4 above)
Comment (artursouza) [reply sdk/src/main/java/io/dapr/actors/client/IActorProxyFactory.java]: https://azure.github.io/azure-sdk/java_design.html#service-client-builder

File: sdk/src/main/java/io/dapr/actors/communication/IActorRequestMessageBody.java (hunk 7 above)
Comment (amanbha) [reply sdk/src/main/java/io/dapr/actors/communication/IActorRequestMessageBody.java]: This interface is not needed for non remoting invocations.

File: sdk/src/main/java/io/dapr/actors/communication/IActorResponseMessageBody.java (hunk 6 above)
Comment (amanbha) [reply sdk/src/main/java/io/dapr/actors/communication/IActorResponseMessageBody.java]: This interface is not needed for non remoting invocations.

File: sdk/src/main/java/io/dapr/actors/client/ActorProxy.java [hunk 8]
```diff
@@ -0,0 +1,81 @@
...
+  public ActorId getActorId() {
+    return actorId;
+  }
+
+  public void setActorId(ActorId actorId) {
+    this.actorId = actorId;
+  }
+
+  public IActorMessageBodyFactory getActorMessageBodyFactory() {
+    return ActorMessageBodyFactory;
```
Comment (amanbha) [inline sdk/src/main/java/io/dapr/actors/client/ActorProxy.java]: This is not needed for non remoting invocations.

File: sdk/src/main/java/io/dapr/actors/client/ActorProxy.java [hunk 9]
```diff
@@ -0,0 +1,81 @@
...
+  }
+
+  public void setActorMessageBodyFactory(IActorMessageBodyFactory ActorMessageBodyFactory) {
```
Comment (amanbha) [inline sdk/src/main/java/io/dapr/actors/client/ActorProxy.java]: This is not needed for non remoting invocations

File: sdk/src/main/java/io/dapr/actors/communication/IActorMessageBodyFactory.java [hunk 10]
```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT License.
+ */
+package io.dapr.actors.communication;
+
+/**
+ * Defines the interface that must be implemented for providing factory for
+ * creating actor request body and response body objects.
```
Comment (amanbha) [inline sdk/src/main/java/io/dapr/actors/communication/IActorMessageBodyFactory.java]: This is not needed for non remoting invocations

File: sdk/src/main/java/io/dapr/actors/communication/client/ActorRemotingClient.java [hunk 11]
```diff
@@ -0,0 +1,12 @@
+/*
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT License.
+ */
+package io.dapr.actors.communication.client;
+
+public class ActorRemotingClient {
```
Comment (amanbha) [inline sdk/src/main/java/io/dapr/actors/communication/client/ActorRemotingClient.java]: This is not needed for non remoting invocations

File: sdk/src/main/java/io/dapr/actors/client/ActorProxy.java (hunk 3 above)
Comment (ji11er) [reply sdk/src/main/java/io/dapr/actors/client/ActorProxy.java]: sorry, missed that

**Pull request closed without merging**